url2bc31e3489_ramonli_RabbitMQResearch_tgz-pJ8-com_mpos_lottery_te_thirdpartyservice_amqp_protobuf_CancellationMessagePulisherJ8/src/com/mpos/lottery/te/thirdpartyservice/amqp/TeTransactionMessage.java

```diff
@@ -0,0 +1,6710 @@
+        com.google.protobuf.ByteString b = 
+            com.google.protobuf.ByteString.copyFromUtf8(
+                (java.lang.String) ref);
+        serialNo_ = b;
+        return b;
+      } else {
+        return (com.google.protobuf.ByteString) ref;
+      }
+    }
+
+    // required .GameInstance game_instance = 5;
+    public static final int GAME_INSTANCE_FIELD_NUMBER = 5;
+    private com.mpos.lottery.te.thirdpartyservice.amqp.TeTransactionMessage.GameInstance gameInstance_;
+    /**
+     * <code>required .GameInstance game_instance = 5;</code>
+     */
+    public boolean hasGameInstance() {
+      return ((bitField0_ & 0x00000010) == 0x00000010);
+    }
+    /**
+     * <code>required .GameInstance game_instance = 5;</code>
+     */
+    public com.mpos.lottery.te.thirdpartyservice.amqp.TeTransactionMessage.GameInstance getGameInstance() {
+      return gameInstance_;
+    }
+    /**
+     * <code>required .GameInstance game_instance = 5;</code>
+     */
+    public com.mpos.lottery.te.thirdpartyservice.amqp.TeTransactionMessage.GameInstanceOrBuilder getGameInstanceOrBuilder() {
+      return gameInstance_;
+    }
+
+    // repeated .Sale.Entry entries = 6;
+    public static final int ENTRIES_FIELD_NUMBER = 6;
+    private java.util.List<com.mpos.lottery.te.thirdpartyservice.amqp.TeTransactionMessage.Sale.Entry> entries_;
+    /**
+     * <code>repeated .Sale.Entry entries = 6;</code>
+     */
+    public java.util.List<com.mpos.lottery.te.thirdpartyservice.amqp.TeTransactionMessage.Sale.Entry> getEntriesList() {
+      return entries_;
+    }
+    /**
+     * <code>repeated .Sale.Entry entries = 6;</code>
+     */
+    public java.util.List<? extends com.mpos.lottery.te.thirdpartyservice.amqp.TeTransactionMessage.Sale.EntryOrBuilder> 
+        getEntriesOrBuilderList() {
+      return entries_;
+    }
+    /**
+     * <code>repeated .Sale.Entry entries = 6;</code>
+     */
+    public int getEntriesCount() {
+      return entries_.size();
+    }
+    /**
+     * <code>repeated .Sale.Entry entries = 6;</code>
+     */
+    public com.mpos.lottery.te.thirdpartyservice.amqp.TeTransactionMessage.Sale.Entry getEntries(int index) {
+      return entries_.get(index);
+    }
+    /**
+     * <code>repeated .Sale.Entry entries = 6;</code>
+     */
+    public com.mpos.lottery.te.thirdpartyservice.amqp.TeTransactionMessage.Sale.EntryOrBuilder getEntriesOrBuilder(
+        int index) {
+      return entries_.get(index);
+    }
+
+    private void initFields() {
+      transaction_ = com.mpos.lottery.te.thirdpartyservice.amqp.TeTransactionMessage.Transaction.getDefaultInstance();
+      multiDraw_ = 0;
+      totalAmount_ = "";
+      serialNo_ = "";
+      gameInstance_ = com.mpos.lottery.te.thirdpartyservice.amqp.TeTransactionMessage.GameInstance.getDefaultInstance();
+      entries_ = java.util.Collections.emptyList();
+    }
+    private byte memoizedIsInitialized = -1;
+    public final boolean isInitialized() {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 11. By comparison, 98% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test.

We recommend that you simplify this method or break it into multiple methods. For example, consider extracting the code block on lines 4617-4622 into a separate method.